src/main/java/seedu/address/logic/parser/find/FindCommandParser.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.parser.find;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_ITEM_TYPE;
+import static seedu.address.logic.parser.util.GeneralParserUtil.getCommandDetails;
+import static seedu.address.logic.parser.util.GeneralParserUtil.getItemType;
+import static seedu.address.logic.parser.util.GeneralParserUtil.isValidItemType;
+import static seedu.address.model.util.ItemUtil.APPLICATION_ALIAS;
+import static seedu.address.model.util.ItemUtil.COMPANY_ALIAS;
+import static seedu.address.model.util.ItemUtil.PROFILE_ALIAS;
+
+import seedu.address.logic.commands.find.FindCommand;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * todo javadocs
+ */
+public class FindCommandParser implements Parser<FindCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindCommand parse(String args) throws ParseException {
+
+        String itemType = getItemType(args, FindCommand.MESSAGE_USAGE);
+        String commandDetails = getCommandDetails(args);
+
+        isValidItemType(itemType);
+        checkCommandDetailsIsNotBlank(commandDetails, itemType);
+
+        switch (itemType) {
+        case COMPANY_ALIAS:
+            // todo shawn
+            //return new FindCompanyCommand(commandDetails);
+            return null;
+        case APPLICATION_ALIAS:
+            // todo sean
+            //return new FindApplicationCommand(commandDetails);
+            return null;
+        case PROFILE_ALIAS:
+            // todo isaac
+            //return new FindProfileCommand(commandDetails);
+            return null;
+        default:
+            // Invalid item type
+            throw new ParseException(MESSAGE_INVALID_ITEM_TYPE);
+        }
+    }
+
+    /**
+     * todo javadocs
+     */
+    private void checkCommandDetailsIsNotBlank(String commandDetails, String itemType) throws ParseException {
+        if (commandDetails.isBlank()) {
+            switch (itemType) {
+            case COMPANY_ALIAS:
+                //throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                //    FindCompanyCommand.MESSAGE_USAGE));
+
+            case APPLICATION_ALIAS:
+                //throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                //   FindApplicationCommand.MESSAGE_USAGE));
+
+            case PROFILE_ALIAS:
+                //throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                //   FindProfileCommand.MESSAGE_USAGE));
+
+            default:
+                // Invalid item type
+                throw new ParseException(MESSAGE_INVALID_ITEM_TYPE);
+            }
```

Comment:
I think this command should definitely be shifted to the GeneralParserUtil class, since it is being used in many of the command parsers

Reply:
Yes, i agree as well. I will help to reformat the rest of the parser

Reply:
While trying to reformat, I realized that We would need to know the Type of command in order to get its `MESSAGE_USAGE`. If we were to shift it to `GeneralParserUtil`, we would need to determine the Type of command and pass it as a a parameter into the static method in `GeneralParserUtil`. However, this determining of the type of command is already present in the `parse` command, as such we might be writing two `switch cases`. However, if we would want to write only one `switch cases`, we might have to change how the code would like like.

Reply:
Also from the way we coded previously, the switch case in the parse would never go to the default case, we can do an assert false there?

Reply:
However, a method that is possible is that we will have to shift that `checkCommandDetailsIsNotBlank` into each of the cases in the switch method as well as introducing an extra parameter.
```
    public FindCommand parse(String args) throws ParseException {

        String itemType = getItemType(args, FindCommand.MESSAGE_USAGE);
        String commandDetails = getCommandDetails(args);

        isValidItemType(itemType);
        switch (itemType) {
        case COMPANY_ALIAS:
            checkCommandDetailsIsNotBlank(commandDetails, itemType, FindCompanyCommand.MESSAGE_USAGE);
            //return new FindCompanyCommand(commandDetails);
            return null;
        case APPLICATION_ALIAS:
            checkCommandDetailsIsNotBlank(commandDetails, itemType, FindApplicationCommand.MESSAGE_USAGE);
            //return new FindApplicationCommand(commandDetails);
            return null;
        case PROFILE_ALIAS:
            checkCommandDetailsIsNotBlank(commandDetails, itemType, FindProfileCommand.MESSAGE_USAGE);
            //return new FindProfileCommand(commandDetails);
            return null;
        default:
            // Invalid item type
            throw new ParseException(MESSAGE_INVALID_ITEM_TYPE);
        }
    }

Reply:
If you are putting the message into the checkCommandDetails already, we can just throw the exception with the correct message? There's no need to do the switch case inside that method anymore.

Reply:
Sorry, I don't quite get what do you mean as throw the exception with the correct message.